src/tools/compiletest/src/runtest.rs

```diff
@@ -1879,22 +1879,92 @@ actual:\n\
     fn run_rustdoc_test(&self) {
         assert!(self.revision.is_none(), "revisions not relevant here");
 
-        let out_dir = self.output_base_name();
-        let _ = fs::remove_dir_all(&out_dir);
-        self.create_dir_racy(&out_dir);
+        if self.props.compile_flags.contains(&"--test".to_owned()) &&
+           self.props.check_stdout == true {
+            self.check_rustdoc_test_option();
+        } else {
+            let out_dir = self.output_base_name();
+            let _ = fs::remove_dir_all(&out_dir);
+            self.create_dir_racy(&out_dir);
 
-        let proc_res = self.document(&out_dir);
-        if !proc_res.status.success() {
-            self.fatal_proc_rec("rustdoc failed!", &proc_res);
+            let proc_res = self.document(&out_dir);
+            if !proc_res.status.success() {
+                self.fatal_proc_rec("rustdoc failed!", &proc_res);
+            }
+            let root = self.find_rust_src_root().unwrap();
+
+            let res = self.cmd2procres(Command::new(&self.config.docck_python)
```

Comment:
The dispatch of what to test should be here. Running `self.document` above should be a common code path, and then whether to execute docck or the check below is where the branch happens.